### DISCUSSION TITLE
Importing Loring RST_LOG csv files

### DISCUSSION THREAD

**Discussion by crankhousecoffee** [artisan-roaster-scope/artisan#897] [Q&A]
Hi folks, 

Does anyone know if it's possible to import a CSV file into Artisan generated by the Loring roast log system (the one that automatically emails you at the end of a roast if set) ?

I've tried the obvious Import->Artisan CSV and a few of the other built in options but no joy so far.

Appreciate any guidance please.
Best regards
Dave

**Comment by MAKOMO** [comment]
Dave, for now there is no Loring CSV import. Just because we didn't know such files exist at all. This has not stay like this. Could you attach one such file here (by drag-and-drop to the message window; you might have to rename it to *.txt) for let me take a look?

Thanks,
Marko

**Comment by crankhousecoffee** [comment]
Hi Marko,

Thanks for getting back to me on this.

Here’s the CSV - I’ve forwarded the entire auto-generated email for you also so you know what the Loring generates and emails at the end of every roast.


> Begin forwarded message:
> 
> From: "Loring X2 HMI" ***@***.***>
> Subject: Roast Log from Loring Smart Roast
> Date: 14 June 2022 at 14:39:55 BST
> To: ***@***.***>
> 
> Batch ID: 202206140023090032
> Date: 6/14/2022 2:39:54 PM
> Recipe Name: COLRISARALDADECAF7KG215
> Product Name: Typica
> Bean Drop Temperature: 220.0 °C
> Roast End Point: 215.6 °C
> Roast Time: 12:18
> Air Cool Deg before EOR: 2.8 °C
> Air Cool Time Duration: 30
> Air Cool Enabled (1 = On, 0 = Off): 0
> 
> Roast Data Comments:
> 
> 
> 
> 
> Initial Burner Setting (IBS): 95
> Batch Size: 12.0 kg
> Roast Type: 0
> Turn Around Point (TAP) Time: 00:45
> Color Change: 05:27
> 1st Crack: 09:22
> 2nd Crack: 00:00
> 
> 
> Attached Roast Log Data Column Headings
> 	A. Sample ID
> 	B. Date/Time
> 	C. Roasting (1 = On, 0 = Off)
> 	D. Roast Time (Seconds)
> 	E. Endpoint Temperature (°C x 10)
> 	F. Burner Percentage (%) (0-100% x 100)
> 	G. Bean Temperature (°C x 10)
> 	H. Bean Temperature Rate of Rise (RoR) (°C x 10)
> 	I. Inlet Air Temperature (°C x 10)
> 	J. Return Air Temperature (°C x 10)
> 	K. Stack Temperature (°C x 10)
> 	L. Profile or Baseline (°C x 10)
> 	M. Reserved
> 	N. Roast Mode (1 Manual, 2 Burner, 3 Profile)
> 	O. Degrees C or Degrees F (0 = °F, 1 = °C)
> 	P. Color Change Time Mark (1 = On, 0 = Off)
> 	Q. 1st Crack Time Mark (1 =  On, 0 = Off)
> 	R. 2nd Crack time Mark (1 =  On, 0 = Off) 
> 
> Version Information:
> Screen:
> 	1009790 2.3.7264.29847
> CF Rev: 2.4.7264.19905
> HMI OBJ Rev: 2.4.7264.19905
> 
> PLC Part No: 1009564
> PLC Version: 2.3.7264.29847
> Roaster ID: 002309

And renamed just in case..



Best regards
Dave




07588 020288

***@***.***

  

> On 15 Jun 2022, at 14:52, Marko Luther ***@***.***> wrote:
> 
> 
> Dave, for now there is no Loring CSV import. Just because we didn't know such files exist at all. This has not stay like this. Could you attach one such file here (by drag-and-drop to the message window; you might have to rename it to *.txt) for let me take a look?
> 
> Thanks,
> Marko
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/artisan-roaster-scope/artisan/discussions/897#discussioncomment-2956182>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AZUPZQP7UIMA5SFJVJRM2C3VPHNYZANCNFSM5Y3ICFPQ>.
> You are receiving this because you authored the thread.
> 


Id,Time,RoastingOnOff,RoastTimeSeconds,EndPointTemperature,BurnerPercentage,BeanTemperature,BeanTemperatureRateOfRise,InletAirTemperature,ReturnAirTemperature,StackTemperature,CurveTemperatureBaselineOrProfile,Reserved,RoastMode_1_2_3,DegreesCelciusOnOff,ColorChangeOnOff,FirstCrackOnOff,SecondCrackOnOff
1,6/14/2022 1:27:18 PM,1,2,2156,2000,1899,0,2632,2220,5666,1768,0,1,1,0,0,0
2,6/14/2022 1:27:24 PM,1,8,2156,2000,1457,0,2655,2063,5660,1387,0,1,1,0,0,0
3,6/14/2022 1:27:30 PM,1,13,2156,2000,1130,-29683,2660,1893,5670,1076,0,1,1,0,0,0
4,6/14/2022 1:27:36 PM,1,19,2156,5527,861,-29683,2640,1708,5688,843,0,1,1,0,0,0
5,6/14/2022 1:27:42 PM,1,26,2156,9493,713,-29683,2623,1591,5730,705,0,1,1,0,0,0
6,6/14/2022 1:27:48 PM,1,32,2156,9493,642,-29683,2612,1556,5863,640,0,1,1,0,0,0
7,6/14/2022 1:27:54 PM,1,38,2156,9493,618,-29683,2612,1553,6025,620,0,1,1,0,0,0
8,6/14/2022 1:28:00 PM,1,43,2156,9493,623,-29345,2615,1557,6140,618,0,1,1,0,0,0
9,6/14/2022 1:28:06 PM,1,49,2156,9493,642,59,2619,1571,6267,631,0,1,1,0,0,0
10,6/14/2022 1:28:12 PM,1,56,2156,9493,673,208,2623,1594,6384,654,0,1,1,0,0,0
11,6/14/2022 1:28:18 PM,1,62,2156,9493,705,288,2627,1612,6473,678,0,1,1,0,0,0
12,6/14/2022 1:28:24 PM,1,68,2156,9493,741,339,2630,1633,6551,702,0,1,1,0,0,0
13,6/14/2022 1:28:30 PM,1,74,2156,9493,778,346,2632,1655,6620,727,0,1,1,0,0,0
14,6/14/2022 1:28:36 PM,1,79,2156,9493,806,344,2633,1671,6663,748,0,1,1,0,0,0
15,6/14/2022 1:28:42 PM,1,85,2156,9493,841,347,2636,1687,6710,781,0,1,1,0,0,0
16,6/14/2022 1:28:48 PM,1,91,2156,9493,877,346,2638,1708,6751,817,0,1,1,0,0,0
17,6/14/2022 1:28:54 PM,1,98,2156,9493,915,343,2641,1728,6787,856,0,1,1,0,0,0
18,6/14/2022 1:29:00 PM,1,104,2156,9493,947,346,2643,1743,6818,889,0,1,1,0,0,0
19,6/14/2022 1:29:06 PM,1,110,2156,9493,981,328,2646,1760,6845,917,0,1,1,0,0,0
20,6/14/2022 1:29:12 PM,1,116,2156,9493,1013,322,2647,1776,6866,940,0,1,1,0,0,0
21,6/14/2022 1:29:18 PM,1,121,2156,9493,1037,308,2650,1789,6878,955,0,1,1,0,0,0
22,6/14/2022 1:29:24 PM,1,127,2156,9493,1065,296,2652,1804,6895,975,0,1,1,0,0,0
23,6/14/2022 1:29:30 PM,1,133,2156,9493,1093,275,2655,1821,6908,997,0,1,1,0,0,0
24,6/14/2022 1:29:36 PM,1,140,2156,9493,1121,267,2657,1838,6918,1023,0,1,1,0,0,0
25,6/14/2022 1:29:42 PM,1,146,2156,9493,1147,258,2661,1850,6928,1052,0,1,1,0,0,0
26,6/14/2022 1:29:48 PM,1,152,2156,8995,1174,259,2663,1858,6936,1080,0,1,1,0,0,0
27,6/14/2022 1:29:54 PM,1,157,2156,8995,1195,257,2666,1869,6938,1100,0,1,1,0,0,0
28,6/14/2022 1:30:00 PM,1,163,2156,8995,1218,242,2668,1875,6936,1122,0,1,1,0,0,0
29,6/14/2022 1:30:06 PM,1,170,2156,8995,1242,227,2672,1886,6937,1142,0,1,1,0,0,0
30,6/14/2022 1:30:12 PM,1,176,2156,8995,1265,220,2675,1898,6935,1161,0,1,1,0,0,0
31,6/14/2022 1:30:18 PM,1,182,2156,8995,1287,215,2677,1906,6933,1179,0,1,1,0,0,0
32,6/14/2022 1:30:24 PM,1,188,2156,8995,1307,215,2681,1914,6931,1199,0,1,1,0,0,0
33,6/14/2022 1:30:30 PM,1,193,2156,8995,1322,200,2685,1921,6927,1217,0,1,1,0,0,0
34,6/14/2022 1:30:36 PM,1,199,2156,8995,1342,190,2688,1932,6926,1239,0,1,1,0,0,0
35,6/14/2022 1:30:42 PM,1,205,2156,8995,1360,184,2692,1940,6926,1262,0,1,1,0,0,0
36,6/14/2022 1:30:48 PM,1,212,2156,8995,1381,180,2698,1947,6923,1285,0,1,1,0,0,0
37,6/14/2022 1:30:54 PM,1,218,2156,8995,1398,177,2702,1958,6923,1306,0,1,1,0,0,0
38,6/14/2022 1:31:00 PM,1,224,2156,8995,1415,166,2707,1967,6921,1324,0,1,1,0,0,0
39,6/14/2022 1:31:06 PM,1,230,2156,8995,1428,161,2711,1976,6918,1340,0,1,1,0,0,0
40,6/14/2022 1:31:12 PM,1,236,2156,8995,1444,161,2715,1985,6921,1357,0,1,1,0,0,0
41,6/14/2022 1:31:18 PM,1,242,2156,8995,1460,155,2720,1995,6926,1375,0,1,1,0,0,0
42,6/14/2022 1:31:24 PM,1,248,2156,8995,1476,151,2725,2005,6928,1392,0,1,1,0,0,0
43,6/14/2022 1:31:30 PM,1,254,2156,8995,1491,150,2728,2012,6934,1411,0,1,1,0,0,0
44,6/14/2022 1:31:36 PM,1,259,2156,8995,1502,141,2731,2018,6940,1425,0,1,1,0,0,0
45,6/14/2022 1:31:43 PM,1,265,2156,8995,1518,145,2733,2026,6945,1442,0,1,1,0,0,0
46,6/14/2022 1:31:49 PM,1,272,2156,8995,1532,146,2737,2033,6951,1460,0,1,1,0,0,0
47,6/14/2022 1:31:55 PM,1,278,2156,8995,1546,137,2740,2043,6953,1475,0,1,1,0,0,0
48,6/14/2022 1:32:01 PM,1,284,2156,8995,1560,138,2743,2055,6953,1489,0,1,1,0,0,0
49,6/14/2022 1:32:07 PM,1,290,2156,8995,1572,132,2746,2062,6954,1502,0,1,1,0,0,0
50,6/14/2022 1:32:13 PM,1,295,2156,8995,1582,122,2748,2068,6952,1513,0,1,1,0,0,0
51,6/14/2022 1:32:19 PM,1,302,2156,8995,1596,125,2752,2076,6953,1527,0,1,1,0,0,0
52,6/14/2022 1:32:25 PM,1,308,2156,8995,1608,124,2755,2085,6951,1542,0,1,1,0,0,0
53,6/14/2022 1:32:31 PM,1,314,2156,8995,1620,118,2757,2091,6953,1557,0,1,1,0,0,0
54,6/14/2022 1:32:37 PM,1,320,2156,8995,1631,111,2759,2097,6960,1572,0,1,1,0,0,0
55,6/14/2022 1:32:43 PM,1,325,2156,8995,1640,107,2761,2105,6963,1583,0,1,1,0,0,0
56,6/14/2022 1:32:49 PM,1,332,2156,8995,1653,115,2764,2112,6965,1598,0,1,1,1,0,0
57,6/14/2022 1:32:55 PM,1,338,2156,8995,1664,109,2766,2116,6972,1611,0,1,1,1,0,0
58,6/14/2022 1:33:01 PM,1,344,2156,8995,1675,107,2769,2123,6975,1624,0,1,1,1,0,0
59,6/14/2022 1:33:07 PM,1,350,2156,8995,1686,102,2771,2129,6976,1637,0,1,1,1,0,0
60,6/14/2022 1:33:13 PM,1,355,2156,8995,1694,108,2773,2135,6976,1646,0,1,1,1,0,0
61,6/14/2022 1:33:19 PM,1,362,2156,8995,1704,103,2777,2141,6975,1659,0,1,1,1,0,0
62,6/14/2022 1:33:25 PM,1,368,2156,8995,1715,98,2780,2150,6977,1671,0,1,1,1,0,0
63,6/14/2022 1:33:31 PM,1,374,2156,8995,1726,101,2783,2157,6980,1683,0,1,1,1,0,0
64,6/14/2022 1:33:37 PM,1,380,2156,8995,1735,98,2787,2165,6982,1694,0,1,1,1,0,0
65,6/14/2022 1:33:43 PM,1,386,2156,8995,1745,95,2789,2170,6986,1704,0,1,1,1,0,0
66,6/14/2022 1:33:49 PM,1,392,2156,8995,1754,98,2792,2175,6990,1714,0,1,1,1,0,0
67,6/14/2022 1:33:55 PM,1,398,2156,8995,1764,96,2795,2182,6992,1725,0,1,1,1,0,0
68,6/14/2022 1:34:01 PM,1,404,2156,8995,1773,91,2798,2188,6992,1735,0,1,1,1,0,0
69,6/14/2022 1:34:07 PM,1,410,2156,8995,1783,93,2801,2194,6992,1746,0,1,1,1,0,0
70,6/14/2022 1:34:13 PM,1,415,2156,8995,1791,97,2804,2198,6992,1755,0,1,1,1,0,0
71,6/14/2022 1:34:19 PM,1,422,2156,8995,1801,94,2808,2210,6996,1766,0,1,1,1,0,0
72,6/14/2022 1:34:25 PM,1,428,2156,8995,1810,90,2812,2216,6996,1776,0,1,1,1,0,0
73,6/14/2022 1:34:31 PM,1,434,2156,8995,1819,90,2817,2222,6997,1786,0,1,1,1,0,0
74,6/14/2022 1:34:37 PM,1,440,2156,8995,1829,93,2821,2227,7001,1797,0,1,1,1,0,0
75,6/14/2022 1:34:43 PM,1,447,2156,8995,1839,100,2826,2235,7007,1809,0,1,1,1,0,0
76,6/14/2022 1:34:49 PM,1,451,2156,8995,1847,102,2829,2238,7011,1818,0,1,1,1,0,0
77,6/14/2022 1:34:55 PM,1,458,2156,8995,1856,93,2834,2246,7017,1830,0,1,1,1,0,0
78,6/14/2022 1:35:01 PM,1,464,2156,8995,1866,86,2838,2253,7021,1840,0,1,1,1,0,0
79,6/14/2022 1:35:07 PM,1,470,2156,8995,1875,93,2843,2259,7023,1851,0,1,1,1,0,0
80,6/14/2022 1:35:13 PM,1,476,2156,8697,1885,93,2849,2268,7027,1861,0,1,1,1,0,0
81,6/14/2022 1:35:19 PM,1,481,2156,8697,1892,91,2852,2271,7028,1870,0,1,1,1,0,0
82,6/14/2022 1:35:25 PM,1,488,2156,8400,1902,93,2857,2273,7028,1880,0,1,1,1,0,0
83,6/14/2022 1:35:31 PM,1,494,2156,8400,1910,88,2861,2277,7026,1888,0,1,1,1,0,0
84,6/14/2022 1:35:37 PM,1,500,2156,8400,1919,81,2864,2282,7018,1898,0,1,1,1,0,0
85,6/14/2022 1:35:43 PM,1,506,2156,8400,1928,88,2869,2288,7015,1908,0,1,1,1,0,0
86,6/14/2022 1:35:49 PM,1,513,2156,8400,1937,83,2873,2291,7010,1918,0,1,1,1,0,0
87,6/14/2022 1:35:55 PM,1,517,2156,8400,1944,85,2877,2296,7007,1927,0,1,1,1,0,0
88,6/14/2022 1:36:01 PM,1,524,2156,8094,1953,88,2882,2301,7004,1937,0,1,1,1,0,0
89,6/14/2022 1:36:07 PM,1,530,2156,8094,1962,82,2886,2305,6995,1946,0,1,1,1,0,0
90,6/14/2022 1:36:13 PM,1,536,2156,7797,1971,85,2890,2307,6992,1955,0,1,1,1,0,0
91,6/14/2022 1:36:19 PM,1,543,2156,7499,1980,86,2892,2310,6983,1963,0,1,1,1,0,0
92,6/14/2022 1:36:25 PM,1,547,2156,7499,1987,86,2894,2311,6975,1970,0,1,1,1,0,0
93,6/14/2022 1:36:31 PM,1,554,2156,7194,1995,78,2896,2315,6960,1978,0,1,1,1,0,0
94,6/14/2022 1:36:37 PM,1,560,2156,7194,2002,78,2897,2317,6940,1987,0,1,1,1,0,0
95,6/14/2022 1:36:43 PM,1,566,2156,7194,2010,74,2900,2317,6920,1997,0,1,1,1,1,0
96,6/14/2022 1:36:49 PM,1,573,2156,7194,2016,67,2901,2319,6905,2007,0,1,1,1,1,0
97,6/14/2022 1:36:55 PM,1,579,2156,7194,2022,61,2901,2320,6899,2015,0,1,1,1,1,0
98,6/14/2022 1:37:01 PM,1,584,2156,7194,2027,60,2902,2322,6891,2022,0,1,1,1,1,0
99,6/14/2022 1:37:07 PM,1,590,2156,7194,2032,55,2902,2322,6885,2029,0,1,1,1,1,0
100,6/14/2022 1:37:13 PM,1,596,2156,7194,2037,52,2902,2323,6880,2036,0,1,1,1,1,0
101,6/14/2022 1:37:19 PM,1,603,2156,7194,2043,51,2904,2325,6871,2045,0,1,1,1,1,0
102,6/14/2022 1:37:25 PM,1,607,2156,6896,2047,53,2905,2327,6866,2052,0,1,1,1,1,0
103,6/14/2022 1:37:31 PM,1,614,2156,6896,2051,45,2906,2327,6859,2061,0,1,1,1,1,0
104,6/14/2022 1:37:37 PM,1,620,2156,6896,2054,36,2907,2328,6854,2068,0,1,1,1,1,0
105,6/14/2022 1:37:43 PM,1,626,2156,6896,2058,34,2909,2330,6851,2076,0,1,1,1,1,0
106,6/14/2022 1:37:49 PM,1,632,2156,6896,2061,35,2911,2331,6852,2082,0,1,1,1,1,0
107,6/14/2022 1:37:55 PM,1,639,2156,6896,2065,32,2913,2335,6855,2091,0,1,1,1,1,0
108,6/14/2022 1:38:01 PM,1,643,2156,6896,2068,38,2916,2338,6857,2098,0,1,1,1,1,0
109,6/14/2022 1:38:07 PM,1,650,2156,6599,2072,41,2920,2343,6860,2108,0,1,1,1,1,0
110,6/14/2022 1:38:13 PM,1,656,2156,6293,2077,36,2923,2345,6861,2117,0,1,1,1,1,0
111,6/14/2022 1:38:19 PM,1,663,2156,5996,2081,43,2927,2346,6860,2125,0,1,1,1,1,0
112,6/14/2022 1:38:25 PM,1,669,2156,5996,2086,38,2931,2348,6855,2133,0,1,1,1,1,0
113,6/14/2022 1:38:31 PM,1,675,2156,5698,2090,43,2935,2351,6842,2142,0,1,1,1,1,0
114,6/14/2022 1:38:37 PM,1,680,2156,5698,2094,45,2939,2352,6835,2149,0,1,1,1,1,0
115,6/14/2022 1:38:43 PM,1,686,2156,5393,2099,45,2943,2355,6826,2157,0,1,1,1,1,0
116,6/14/2022 1:38:49 PM,1,692,2156,5393,2103,46,2945,2356,6818,2161,0,1,1,1,1,0
117,6/14/2022 1:38:55 PM,1,699,2156,5281,2108,44,2948,2361,6809,2156,0,1,1,1,1,0
118,6/14/2022 1:39:01 PM,1,704,2156,4798,2114,54,2951,2363,6800,2156,0,1,1,1,1,0
119,6/14/2022 1:39:07 PM,1,710,2156,4493,2119,55,2955,2365,6789,2156,0,1,1,1,1,0
120,6/14/2022 1:39:13 PM,1,716,2156,4493,2123,47,2960,2365,6774,2156,0,1,1,1,1,0
121,6/14/2022 1:39:19 PM,1,723,2156,3994,2129,47,2965,2366,6755,2156,0,1,1,1,1,0
122,6/14/2022 1:39:25 PM,1,729,2156,3994,2135,54,2969,2362,6737,2156,0,1,1,1,1,0
123,6/14/2022 1:39:31 PM,1,734,2156,3696,2138,53,2973,2358,6722,2156,0,1,1,1,1,0
124,6/14/2022 1:39:37 PM,0,738,2156,2535,2143,555,2977,2355,6706,2156,0,0,1,1,1,0
126,6/14/2022 1:39:43 PM,,,,,,,,,,,,,,,,
